### PR TITLE
Fix(eos_designs): Create MLAG peer-group when needed for non-BGP underlays

### LINKED ISSUE #2304: Bug: improper router bgp MLAG iBGP peering when underlay_routing_protocol: OSPF
### Issue Summary

In similar fashion to when the `underlay_routing_protocol: EBGP`, the peer group definition stanza should be rendered.
```
@@ -292,6 +292,15 @@ router bgp <ASN>
    neighbor EVPN-OVERLAY-PEERS ebgp-multihop 3
    neighbor EVPN-OVERLAY-PEERS send-community
    neighbor EVPN-OVERLAY-PEERS maximum-routes 0
+   neighbor MLAG-IPv4-UNDERLAY-PEER peer group
+   neighbor MLAG-IPv4-UNDERLAY-PEER remote-as <ASN>
+   neighbor MLAG-IPv4-UNDERLAY-PEER next-hop-self
+   neighbor MLAG-IPv4-UNDERLAY-PEER description <DESC>
+   neighbor MLAG-IPv4-UNDERLAY-PEER send-community
+   neighbor MLAG-IPv4-UNDERLAY-PEER maximum-routes 12000
+   neighbor MLAG-IPv4-UNDERLAY-PEER route-map RM-MLAG-PEER-IN in
+   neighbor <x.x.x.x> peer group MLAG-IPv4-UNDERLAY-PEER
+   neighbor <x.x.x.x> description <DESC>
    address-family ipv4
       no neighbor EVPN-OVERLAY-PEERS activate
+      neighbor MLAG-IPv4-UNDERLAY-PEER activate
    !
    ```

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Create MLAG peer-group when needed for non-BGP underlays

## Related Issue(s)

Fixes #2304

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Also create MLAG peer group in network services to cover the case of underlay not being bgp.
- Refactor mlag a bit
- Added TODOs for duplicated code.

## How to test

Added coverage in molecule for the issue (first commit) and verifying that it was solved with the fix (second commit).

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/avdstructuredconfig.py [hunk 1]
```diff
@@ -337,10 +385,10 @@ def router_bgp(self):
         if self._mlag_ibgp_origin_incomplete is True:
             peer_group["route_map_in"] = "RM-MLAG-PEER-IN"
 
-        router_bgp["peer_groups"] = {peer_group_name: peer_group}
+        router_bgp_cfg["peer_groups"] = {peer_group_name: peer_group}
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/avdstructuredconfig.py]: in term of consistency between method I am not sure why we use in one place `router_bgp` and in the other `router_bgp_cfg`?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py [hunk 2]
```diff
@@ -17,6 +19,8 @@ class RouterBgpMixin(UtilsMixin):
     Class should only be used as Mixin to a AvdStructuredConfig class
     """
 
+    _configure_bgp_mlag_peer_group: bool
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py]: do you still need this now that it is a cached property?

**Review by gmuloc — APPROVED** [review]
LGTM - two remarks that can be addressed before merging

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/avdstructuredconfig.py (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/avdstructuredconfig.py]: fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py]: removed

**Comment by pvinci-arista** [comment]
LGTM!

**Review by carlbuchmann — APPROVED** [review]
LGTM tested with customer-specific scenario

**Pull request merged**